api/storage/client.go

```diff
@@ -49,6 +50,30 @@ func (c *Client) Show(tags []names.StorageTag) ([]params.StorageInstance, error)
 	return all, allErr.Combine()
 }
 
+// PoolList lists pools according to a given filter.
+// If no filter was provided, this will return a list
+// of all pools.
+func (c *Client) PoolList(types, names []string) ([]params.PoolInstance, error) {
```

Comment:
I'd prefer if we stuck with the verb[-object] convention for API method names, and called this ListPools.